base/errorshow.jl

```diff
@@ -98,10 +98,40 @@ function showerror(io::IO, ex, bt; backtrace=true)
     end
 end
 
+function stacktrace_path(file::Union{Nothing, String}, line::Union{Nothing, Int64})
+    realfile = if !isnothing(file) && file != "" && !startswith(String(file), "REPL")
+        fixup_stdlib_path(String(file))
+    end
+    pathstr = file
+    if !isnothing(pathstr)
+        stacktrace_expand_basepaths() && (pathstr = something(find_source_file(file), file))
+        stacktrace_contract_userdir() && (pathstr = contractuser(pathstr))
+    end
+    linestr = if !isnothing(line) && line > 0
+        string(something(pathstr, ""), ':', line)
+    else
+        something(pathstr, "")
+    end
+    if !isnothing(realfile)
+        flen = ncodeunits(basename(realfile))
+        llen = ncodeunits(string(line))
+        AnnotatedString(linestr,
+                        [(1:ncodeunits(linestr), :link => Filesystem.uripath(realfile)),
+                         (1:ncodeunits(linestr)-flen-llen-1, :face => :julia_stacktrace_location),
+                         (ncodeunits(linestr)-flen-llen:ncodeunits(linestr)-llen-1, :face => :julia_stacktrace_filename),
+                         (ncodeunits(linestr)-llen:ncodeunits(linestr), :face => :julia_stacktrace_fileline)])
+    else
+        AnnotatedString(linestr, [(1:ncodeunits(linestr), :face => :julia_stacktrace_location)])
+    end
+end
+
+stacktrace_path(location::LineNumberNode) =
+    stacktrace_path(if !isnothing(location.file) String(location.file) end, location.line)
```

Comment:
syntax nit: I think it is generally considered syntactically preferable (e.g. consistent) to use `isnothing(location.file) ? nothing : String(location.file)` to avoid the negation and to use `?:` instead of `if end` form, but I don't think it is required to change it by the style guide

Reply:
Yeah, it is sort of just a "when in Rome" custom for Base, since not all reviewers may be equally comfortable with alternative syntaxes, but not expected of any stdlib code to conform to it